script-library/docker-debian.sh

```diff
@@ -179,40 +194,80 @@ else
-if type docker-compose > /dev/null 2>&1; then
-    echo "Docker Compose already installed."
-else
-    TARGET_COMPOSE_ARCH="$(uname -m)"
-    if [ "${TARGET_COMPOSE_ARCH}" = "amd64" ]; then
-        TARGET_COMPOSE_ARCH="x86_64"
+# Install Compose v2 plugin if missing
+if [ "${DOCKER_COMPOSE_PLUGIN_VERSION}" != "none" ]; then
+    if [ -e "/usr/local/lib/docker/cli-plugins/docker-compose" ] || [ -e "/usr/local/libexec/docker/cli-plugins/docker-compose" ] || [ -e "/usr/lib/docker/cli-plugins/docker-compose" ] || [ -e "/usr/libexec/docker/cli-plugins/docker-compose" ]; then
+        echo "(*) Compose v2 plugin already installed. Skipping..."
+    else
+        find_version_from_git_tags DOCKER_COMPOSE_V2_VERSION "https://github.com/docker/compose"
+        echo "(*) Installing Compose v2 plugin ${DOCKER_COMPOSE_PLUGIN_VERSION}..."
+        docker_compose_v2_filename="docker-compose-linux-${target_compose_arch}"
+        curl -fsSL "https://github.com/docker/compose/releases/download/v${DOCKER_COMPOSE_PLUGIN_VERSION}/${docker_compose_v2_filename}" -o /tmp/${docker_compose_v2_filename}
+        curl -fsSL "https://github.com/docker/compose/releases/download/v${DOCKER_COMPOSE_PLUGIN_VERSION}/${docker_compose_v2_filename}.sha256" -o /tmp/${docker_compose_v2_filename}.sha256
+        cd /tmp
+        sha256sum -c /tmp/${docker_compose_v2_filename}.sha256
+        rm -f /tmp/${docker_compose_v2_filename}.sha256
+        mkdir -p /usr/local/lib/docker/cli-plugins/
+        mv -f /tmp/${docker_compose_v2_filename} /usr/local/lib/docker/cli-plugins/docker-compose
+        chmod +x /usr/local/lib/docker/cli-plugins/docker-compose
     fi
-    if [ "${TARGET_COMPOSE_ARCH}" != "x86_64" ]; then
-        # Use pip to get a version that runns on this architecture
-        if ! dpkg -s python3-minimal python3-pip libffi-dev python3-venv > /dev/null 2>&1; then
-            apt_get_update_if_needed
-            apt-get -y install python3-minimal python3-pip libffi-dev python3-venv
-        fi
-        export PIPX_HOME=/usr/local/pipx
-        mkdir -p ${PIPX_HOME}
-        export PIPX_BIN_DIR=/usr/local/bin
-        export PYTHONUSERBASE=/tmp/pip-tmp
-        export PIP_CACHE_DIR=/tmp/pip-tmp/cache
-        pipx_bin=pipx
-        if ! type pipx > /dev/null 2>&1; then
-            pip3 install --disable-pip-version-check --no-warn-script-location  --no-cache-dir --user pipx
-            pipx_bin=/tmp/pip-tmp/bin/pipx
+fi
+
+# Install Docker Compose v1 if not already installed
+if [ "${DOCKER_COMPOSE_V1_VERSION}" != "none" ]; then
+    if type docker-compose > /dev/null 2>&1; then
+        echo "(*) Compose v1 already installed. Skipping..."
+    else
+        if [ "${target_compose_arch}" != "x86_64" ]; then
+            # Use pip to get a version that runns on this architecture
+            echo "(*) Installing Compose v1..."
+            if ! dpkg -s python3-minimal python3-pip libffi-dev python3-venv > /dev/null 2>&1; then
+                apt_get_update_if_needed
```

Comment:
I wonder if here (in another PR) would make sense to call out to the python script?